fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs

```diff
@@ -311,9 +359,259 @@ impl PackFlow for AirGenerationSystemApplication {
+            hp_cabin_fans_flow_demand: MassRate::default(),
+            should_switch_on_bulk_heater: false,
+        }
+    }
+
+    fn update(
+        &mut self,
+        acs_overhead: &impl AirConditioningOverheadShared,
+        bulk_control_is_powered: bool,
+        cabin_temperature: &impl CabinSimulation,
+        cargo_door_open: &impl CargoDoorLocked,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+        vcm_shared: &impl VcmShared,
+        pack_flow_demand: &impl PackFlow,
+    ) {
+        self.fwd_extraction_fan_is_on = vcm_shared.fwd_extraction_fan_is_on();
+        self.fwd_isolation_valve_is_open = vcm_shared.fwd_isolation_valves_open_allowed();
+        self.bulk_control_is_powered = bulk_control_is_powered;
+        self.bulk_extraction_fan_is_on = vcm_shared.bulk_extraction_fan_is_on();
+        self.bulk_isolation_valve_is_open =
+            self.bulk_control_is_powered && vcm_shared.bulk_isolation_valves_open_allowed();
+        self.hp_cabin_fans_are_enabled = vcm_shared.hp_cabin_fans_are_enabled();
+        // The recirculation airflow demand is linked to the fresh airflow demand in order to keep the total airflow constant
+        self.hp_cabin_fans_flow_demand = self.recirculation_flow_determination(
+            acs_overhead,
+            pack_flow_demand.pack_flow_demand(Pack(1)) + pack_flow_demand.pack_flow_demand(Pack(2)),
+        );
+        self.should_switch_on_bulk_heater = self.bulk_heater_on_determination(
+            acs_overhead,
+            cabin_temperature,
+            cargo_door_open,
+            lgciu,
+            vcm_shared,
+        );
+    }
+
+    fn recirculation_flow_determination(
+        &self,
+        acs_overhead: &impl AirConditioningOverheadShared,
+        pack_flow_demand: MassRate,
+    ) -> MassRate {
+        MassRate::new::<kilogram_per_second>(
+            Self::TOTAL_MIXED_AIR_DEMAND
+                * Ratio::from(acs_overhead.flow_selector_position()).get::<ratio>()
+                - pack_flow_demand.get::<kilogram_per_second>(),
+        )
+    }
+
+    fn bulk_heater_on_determination(
+        &self,
+        acs_overhead: &impl AirConditioningOverheadShared,
+        cabin_temperature: &impl CabinSimulation,
+        cargo_door_open: &impl CargoDoorLocked,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+        vcm_shared: &impl VcmShared,
+    ) -> bool {
+        let bulk_heater_on_allowed = (cargo_door_open.aft_cargo_door_locked()
+            || !lgciu.iter().all(|a| a.left_and_right_gear_compressed(true)))
+            && vcm_shared.bulk_duct_heater_on_allowed();
+        let temperature_difference = cabin_temperature.cabin_temperature()[ZoneType::Cargo(2).id()]
+            .get::<degree_celsius>()
+            - acs_overhead
+                .selected_cargo_temperature(ZoneType::Cargo(2))
+                .get::<degree_celsius>();
+        (temperature_difference < -1.
+            || (self.should_switch_on_bulk_heater && temperature_difference < 1.))
```

Comment:
Are you sure this condition is correct? Because only `temperature_difference` determines the result of it.

Reply:
The idea is that the heater switches on when temperature difference is less than -1deg, and switches off when it's above +1deg (this condition is essentially keeping the heater on if it was already on and the temperature difference is between -1 and +1)

Reply:
I didn't see that once it's -1 and the other time +1. My bad.